File: frontend/src/types/ApiResponse.ts
import { Word } from "./Word";

export type ApiResponse = {
    language: string;
    totalRepositoriesCount: number;
    repositoriesAnalysed: number;
    classesAnalysed: number;
    validClasses: number;
    wordsAnalysed: number;
    averageWordsPerClass: number;
    percentageOfValidClasses: number;
    words: Word[];
};

File: backend/src/main/kotlin/pl/pszumanski/classnameanalyser/dto/api/ApiResponse.kt
package pl.pszumanski.classnameanalyser.dto.api

import pl.pszumanski.classnameanalyser.values.SupportedLanguage

data class ApiResponse(val language: SupportedLanguage, val totalRepositoriesCount: Int, val repositoriesAnalysed: Int, val classesAnalysed: Int,
                       val validClasses: Int, val wordsAnalysed: Int, val averageWordsPerClass: Double,
                       val percentageOfValidClasses: Double, val words: Map<String, Int>)
